server/src/test/java/org/elasticsearch/env/NodeRepurposeCommandIT.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.env;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.hamcrest.Matcher;
+
+import java.util.stream.IntStream;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+import static org.mockito.Matchers.contains;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class NodeRepurposeCommandIT extends ESIntegTestCase {
```

Comment:
Since it's an integration test, I think we need to be able to verify that the document that we have indexed is lost. For example, you may start one master-only node and one data-only node and then repurpose data-only to be master-only or no-master no-data. In this case, when the repurposed node joins the cluster, the index will be re-created but the document is gone.  
Also due to randomization, `executeRepurposeCommand` is a bit hard to follow. I think here we can stop a particular node using `nameFilter` and explicitly run repurpose tool successfully on this node and unsuccessfully on another node.

Reply:
Regarding checking for lost data, I added a check that we loose the index completely in f5396b18519b6d8115852673b43fe692dade310c , I think that should suffice, do you agree?

Commit 85cf0e65edcaefb2f80280dd936b7c24839e0eae fixes test to stop one node, run the tool successfully on that node and unsuccessfully on the other node.

Reply:
I'm not sure how the test works, but after you restart two nodes - they should not form the cluster, because there is no longer the master node. It'd nice to figure out why ensureGreen returns successfully in this case.
But in general, I think, having one dedicated master-eligible node and one data node and repurposing of data node (mentioned in the comment) would be easier to follow.

Reply:
@andrershov this works because `InternalTestCluster` maintains the voting configuration in `InternalTestCluster.excludeMasters`, which takes the stopping node of the voting configuration. This is used by several other tests (is the default for `InternalTestCluster`), for instance I think `GatewayIndexStateIT.testIndexDeletionWhenNodeRejoins` depends on the same.

Reply:
Changed to one master only and one data only node in 7a0712c9e2f6df80e83d7158ce99b27cf50c88c3.

Reply:
Sorry, the test needed more adjustments, fixed in 4cbf288dab44ebb7422437fcccefeb67790797a6

I somehow ran the wrong test 100 times ;-)